lib/rules/no-danger.js

```diff
@@ -55,13 +56,33 @@ module.exports = {
 
     messages,
 
-    schema: [],
+    schema: [{
+      type: 'object',
+      properties: {
+        customComponentNames: {
+          items: {
+            type: 'string',
+          },
+          minItems: 1,
+          type: 'array',
+          uniqueItems: true,
+        },
+      },
+    }],
   },
 
   create(context) {
+    const configuration = context.options[0] || {};
+    const customComponentNames = configuration.customComponentNames || [];
+
     return {
       JSXAttribute(node) {
-        if (jsxUtil.isDOMComponent(node.parent) && isDangerous(node.name.name)) {
+        const functionName = node.parent.name.name;
+
+        const enableCheckingCustomComponent = customComponentNames.length > 0
+          && customComponentNames.some((name) => minimatch(functionName, name));
```

Comment:
```suggestion
        const enableCheckingCustomComponent = customComponentNames.some((name) => minimatch(functionName, name));
```
.some already handles empty lists :-)